src/Interpreters/Context.cpp

```diff
@@ -1100,6 +1101,7 @@ void Context::applySettingsChanges(const SettingsChanges & changes)
     auto lock = getLock();
     for (const SettingChange & change : changes)
         applySettingChange(change);
+    applySettingsQuirks(settings);
```

Comment:
Maybe add this line into `Client.cpp` too?

Reply:
But client does not deal with async stuff, everything is done by the server.
And besides it does not know version of the kernel on the server (and not sure that it worth adding since only builtin will deal with it)